### PR TITLE
More reliable scheduler result query API

### PR DESCRIPTION
**Pull request by tbak**

### CODE REVIEW

File: titus-server-master/src/main/java/com/netflix/titus/master/scheduler/SchedulingService.java [hunk 1]
```diff
@@ -87,4 +89,17 @@ void registerTaskQListAction(
     void registerTaskFailuresAction(
             String taskId, com.netflix.fenzo.functions.Action1<List<TaskAssignmentResult>> action
     ) throws IllegalStateException;
+
+    /**
+     * Returns the last know scheduling result for a task.
```
Comment (corindwyer) [inline titus-server-master/src/main/java/com/netflix/titus/master/scheduler/SchedulingService.java]: know -> known

File: titus-server-master/src/test/java/com/netflix/titus/master/jobmanager/service/integration/scenario/StubbedSchedulingService.java [hunk 2]
```diff
@@ -93,4 +96,14 @@ public void registerTaskQListAction(com.netflix.fenzo.functions.Action1<Map<Task
     @Override
     public void registerTaskFailuresAction(String taskId, com.netflix.fenzo.functions.Action1<List<TaskAssignmentResult>> action) throws IllegalStateException {
     }
+
+    @Override
+    public Optional<SchedulingResultEvent> findLastSchedulingResult(String taskId) {
+        throw new IllegalStateException("not implemented");
```
Comment (corindwyer) [inline titus-server-master/src/test/java/com/netflix/titus/master/jobmanager/service/integration/scenario/StubbedSchedulingService.java]: UnsupportedOperationException?

File: titus-server-master/src/test/java/com/netflix/titus/master/jobmanager/service/integration/scenario/StubbedSchedulingService.java [hunk 3]
```diff
@@ -93,4 +96,14 @@ public void registerTaskQListAction(com.netflix.fenzo.functions.Action1<Map<Task
...
+    }
+
+    @Override
+    public Observable<SchedulingResultEvent> observeSchedulingResults(String taskId) {
+        return Observable.error(new IllegalStateException("not implemented"));
```
Comment (corindwyer) [inline titus-server-master/src/test/java/com/netflix/titus/master/jobmanager/service/integration/scenario/StubbedSchedulingService.java]: UnsupportedOperationException?

File: titus-server-gateway/src/main/java/com/netflix/titus/gateway/service/v3/SchedulerService.java [hunk 4]
```diff
@@ -64,4 +65,17 @@
      * @throws SchedulerException {@link SchedulerException.ErrorCode#SystemSelectorNotFound} if the system selector is not found
      */
     Completable deleteSystemSelector(String id);
+
+    /**
+     * Returns the last know scheduling result for a task.
```
[the remark this answers is not in the document]
Comment (tbak) [reply titus-server-gateway/src/main/java/com/netflix/titus/gateway/service/v3/SchedulerService.java]: Fixed

File: titus-server-master/src/main/java/com/netflix/titus/master/scheduler/SchedulingService.java (hunk 1 above)
Comment (tbak) [reply titus-server-master/src/main/java/com/netflix/titus/master/scheduler/SchedulingService.java]: Fixed

File: titus-server-master/src/test/java/com/netflix/titus/master/jobmanager/service/integration/scenario/StubbedSchedulingService.java (hunk 2 above)
Comment (tbak) [reply titus-server-master/src/test/java/com/netflix/titus/master/jobmanager/service/integration/scenario/StubbedSchedulingService.java]: Changed

File: titus-server-master/src/test/java/com/netflix/titus/master/jobmanager/service/integration/scenario/StubbedSchedulingService.java (hunk 3 above)
Comment (tbak) [reply titus-server-master/src/test/java/com/netflix/titus/master/jobmanager/service/integration/scenario/StubbedSchedulingService.java]: Changed

**Pull request merged**